### PR TITLE
Simplify build

### PR DESCRIPTION
**Pull request by impressiver**
Modified and added a top-level Makefile, which keep project configuration in one place and compiles everything with a single command. I'm running OS X, but this should work the same for win/linux/osx.

I wasn't sure what to do for default paths. I made them all relative to the project, but there's not really a "defacto" place for both the official sdk and/or esp-open-sdk. The esp8266.com folks use `/opt/Espressif` in their installs/examples, and esp-open-sdk keeps things relative to wherever you checked out the repo. 

Any thoughts/opinions?

### CODE REVIEW

**Comment by impressiver** [comment]
Ignore the tab formatting for now, I'll get that cleaned up

File: rboot-sampleproject/main.c [hunk 1]
```diff
@@ -18,9 +18,6 @@
 
 static os_timer_t network_timer;
 
-void ICACHE_FLASH_ATTR user_rf_pre_init(){
```
Comment (impressiver) [inline rboot-sampleproject/main.c:21]: This was causing a duplicate definition compilation error w/ the latest open sdk. If you think it's necessary, I can figure out a way to make it happy w/ or w/o.

Comment (raburton) [inline rboot-sampleproject/main.c:21]: It's needed for the official sdk.

**Comment by raburton** [comment]
I like the idea, but overall I think it's made it more complicated than it needs to be. I'll push a version based on your parent makefile, but is a bit simpler. It should still do what you want it to though.

**Pull request closed without merging**